src/adapter/src/coord/message_handler.rs

```diff
@@ -179,10 +183,10 @@ impl Coordinator {
         if let Err(err) = self.catalog_transact(None, ops).await {
             tracing::warn!("Failed to update storage metrics: {:?}", err);
         }
-        self.schedule_storage_usage_collection();
+        self.schedule_storage_usage_collection().await;
     }
 
-    pub fn schedule_storage_usage_collection(&self) {
+    pub async fn schedule_storage_usage_collection(&self) {
```

Comment:
might be nice to keep this non-async so it's obvious that it doesn't do anything expensive. which I think could be as easy as passing in the ts as an arg?